### ISSUE TITLE
Have a more robust way of loading files

### ISSUE THREAD

**Issue by arturmuller** [saamo/api-doc-test#4]
Right now this package relies on the the fact that people will first specify a request body and than a response body. This is not very robust. There are several problems:

- What if you accidentally switch req and res around?
- What if your comments don't match (you have more, or less, comments than you have files?)
- What if you want to use the comments for something else entirely?

There is probably more.

My point it, whether the file is a request or a response should be handled in a more robust way. One idea we already discussed was to have comments inside the files, but that would make them invalid JSON, so not sure if that is a good idea. Using them as JS 'modules' with `module.exports` would fix this, but it seems unclean. I like the idea of having my mocks as JSON files.

Anyways, this is an open problem that should probably be addressed at some point in the future.

**Comment by saamo** [comment]
This is definitely a serious issue because it puts the burden on a programmer unnecessarily. So I have an idea — how about putting all the information into the comment itself?

```js
// Request (application/json) mocks/request.json
// Response 201 (application/json) mocks/response.json
```

This solves all the points you mentioned before and it also has some other implications.

You're not anymore dependent on the exact syntax inside your code (no more `var reqFile = 'mocks/request.json'`). However this allows you to be inconsistent (one file is used for the documentation, the second one is used for the actual test) but on the other hand this also gives you a possibility to have for example a _nicer_ response body for the documentation (but not sure how much of advantage this would be in real).

One huge advantage is that you could specify requests/responses even without testing them. I know this sounds weird but you don't want to test 401 error in every action (for example) but you still want to have this error documented.

```js
// Request (application/json) mocks/request.json
// Response 201 (application/json) mocks/response.json
// Response 401 (application/json) mocks/response.401.json
```

What do you think?

**Comment by arturmuller** [comment]
I think this is a great idea! Really neat. All for implementing this!

**Comment by saamo** [comment]
Solved by 243f6ab.
